crates/router/src/connector/boku/transformers.rs

```diff
@@ -1,107 +1,263 @@
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
         match item.request.payment_method_data.clone() {
-            api::PaymentMethodData::Card(req_card) => {
-                let card = BokuCard {
-                    name: req_card.card_holder_name,
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.request.is_auto_capture()?,
-                };
-                Ok(Self {
-                    amount: item.request.amount,
-                    card,
-                })
+            api_models::payments::PaymentMethodData::Wallet(wallet_data) => {
+                Self::try_from((item, &wallet_data))
             }
-            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+            _ => Err(errors::ConnectorError::NotSupported {
+                message: format!("{:?}", item.request.payment_method_type),
+                connector: "Adyen",
+                payment_experience: api_models::enums::PaymentExperience::RedirectToUrl.to_string(),
+            })?,
+        }
+    }
+}
+
+impl TryFrom<(&types::PaymentsAuthorizeRouterData, &api::WalletData)> for BokuPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        value: (&types::PaymentsAuthorizeRouterData, &api::WalletData),
+    ) -> Result<Self, Self::Error> {
+        let (item, wallet_data) = value;
+        let country = match get_country_code(item) {
+            Some(cn_code) => cn_code.to_string(),
+            None => Err(errors::ConnectorError::MissingRequiredField {
+                field_name: "country",
+            })?,
+        };
+        let payment_method = get_wallet_type(wallet_data)?;
+        let hosted = get_hosted_data(item);
+        let auth_type = BokuAuthType::try_from(&item.connector_auth_type)?;
+        let merchant_item_description = get_item_description(item)?;
+        let payment_data = SingleChargeData {
+            total_amount: item.request.amount,
+            currency: item.request.currency.to_string(),
+            country,
+            merchant_id: auth_type.merchant_id,
+            merchant_transaction_id: Secret::new(item.payment_id.to_string()),
+            merchant_request_id: Uuid::new_v4().to_string(),
+            merchant_item_description,
+            notification_url: item.request.webhook_url.clone(),
+            payment_method,
+            charge_type: BokuChargeType::Hosted.to_string(),
+            hosted,
+        };
+
+        Ok(Self::BeginSingleCharge(payment_data))
+    }
+}
+
+fn get_wallet_type(wallet_data: &api::WalletData) -> Result<String, errors::ConnectorError> {
```

Comment:
Change this to TryFrom

Reply:
Since the tryfrom will be implemented on string, I dont think there is much use changing it